lib/utils/queryable.js

```diff
@@ -1,14 +1,30 @@
 const util = require('util')
 const _data = Symbol('data')
 const _delete = Symbol('delete')
+const _append = Symbol('append')
 
 const sqBracketsMatcher = str => str.match(/(.+)\[([^\]]+)\](.*)$/)
 
 const cleanLeadingDot = str =>
   str && str.startsWith('.') ? str.substr(1) : str
 
+// replaces any occurence of an empty-brackets (e.g: []) with a special
+// Symbol(append) to represent it, this is going to be useful for the setter
+// method that will push values to the end of the array when finding these
+const replaceAppendSymbols = str => {
+  const matchEmptyBracket = str.match(/^(.*)\[\](.*)$/)
+
+  if (matchEmptyBracket) {
+    const [, pre, post] = matchEmptyBracket
+    return [pre, _append, cleanLeadingDot(post)].filter(Boolean)
```

Comment:
```suggestion
  const matchEmptyBracket = str.match(/^(.*)\[\]\.?(.*)$/)

  if (matchEmptyBracket) {
    const [, pre, post] = matchEmptyBracket
    return [pre, _append, post].filter(Boolean)
```

Reply:
oh thanks, yeah I'm terrible with regexes 😂 I'll also apply it to the previous regex so that I can get rid of the `cleanLeadingDot` fn  😊